server/src/main/java/org/elasticsearch/cluster/metadata/AliasMetaData.java

```diff
@@ -194,6 +210,11 @@ public AliasMetaData(StreamInput in) throws IOException {
             searchRouting = null;
             searchRoutingValues = emptySet();
         }
+        if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            writeIndex = in.readOptionalBoolean();
+        } else {
+            writeIndex = null;
+        }
     }
 
     public static Diff<AliasMetaData> readDiffFrom(StreamInput in) throws IOException {
```

Comment:
I have a general question about cluster-state diffs, if you don't mind sharing. We seem to check stream-input version in the general Writeable.Reader, but I do not see any here. I'm not entirely sure why that is the case, shouldn't we also version-guard reading diffs across the wire?

Reply:
oh, nevermind. the diff is just a complete overwrite. not part-by-part, so the wire-version is covered